provider/vpc/diagram.py

```diff
@@ -1,52 +1,122 @@
-from typing import List
+from typing import List, Dict, Optional
 
-from diagrams import Cluster, Diagram
+from shared.common import ResourceEdge, Resource, ResourceDigest
+from shared.diagram import BaseDiagram, Mapsources
 
-from shared.common import Resource
-from shared.diagram import BaseDiagram, Mapsources, PATH_DIAGRAM_OUTPUT
-from shared.error_handler import exception
+PUBLIC_SUBNET = "{public subnet}"
+PRIVATE_SUBNET = "{private subnet}"
+
+
+def to_node_get_aggregated(
+    resource_relation: ResourceEdge, resources: List[Resource]
+) -> Optional[Resource]:
+    for resource in resources:
+        if (
+            " subnet}" in resource.digest.id
+            and resource_relation.to_node.id in resource.name
+        ):
+            return resource
+    return None
+
+
+def aggregate_subnets(groups, group_type, group_name):
+    if group_type in groups:
+        subnet_ids = []
+        for subnet in groups[group_type]:
+            subnet_ids.append(subnet.digest.id)
+        groups[""].append(
+            Resource(
+                digest=ResourceDigest(id=group_type, type="aws_subnet"),
+                name=group_name + ", ".join(subnet_ids),
+            )
+        )
 
 
 class VpcDiagram(BaseDiagram):
+    def __init__(self, name: str, filename: str, vpc_id: str):
+        """
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Multi-line docstring summary should start at the first line (D212)](https://app.codacy.com/gh/Cloud-Architects/cloud-discovery/pullRequest?prid=5655786)